code-challenges.test.js

```diff
@@ -13,47 +13,147 @@
+    // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+    
+    //hardcode what you would pass as an argument and what the 
+    //expected outcomes will be.
+    expect(capitalized(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+    expect(capitalized(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+  })
+})
 
 // b) Create the function that makes the test pass.
+//create a function named 'capitalized' that takes an array as an argument. Use the .map() method to iterate through
+//the array. Use the .charAt() method to take the first character of each value which in this case would be the first
+//letter of each word - set that character to uppercase using .toUpperCase() method. add the additional characters of
+//each value using the .slice() method. We use .sclice(1) to take the remaining characters of each word and return a new
+//array with all the words capitalized.
+const capitalized = (array) => {return array.map((value) => value.charAt(0).toUpperCase()+value.slice(1))}
+//Note: this one was especially difficult to figure out. If there is an easier to understand method please
+//let me know. looking at it makes my eyes hurt.
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+//Create a test for a function named 'firstVowel'.
+describe('firstVowel', () => {
+
+  //Explain how the function is supposed to perfrom.
+  it('takes in a string and logs the index of the first vowel', () => {
+
+    //Variables that will be used for the expect.
+    const vowelTester1 = "learn";
+    // Expected output: 1
+    const vowelTester2 = "academy";
+    // Expected output: 0
+    const vowelTester3 = "challenges";
+    // Expected output: 2
+
+    //hardcode what you would pass as an argument and what the 
+    //expected outcomes will be.
+    expect(firstVowel(vowelTester1 )).toEqual(1)
+    expect(firstVowel(vowelTester2)).toEqual(0)
+    expect(firstVowel(vowelTester3)).toEqual(2)
+  })
+})
 
 // b) Create the function that makes the test pass.
+//Note: I have attempted to use filter and map methods but I can't seem to figure it out,
+//if there is a way please let me know.
+
+//Create a function named firstVowel that takes a string as a parameter.
+const firstVowel = (string) => {
+  //create a for loop that iterates through the given strings.
+  for (let i = 0; i < string.length; i++){
+    //If, while iterating, i is the same as a vowel we return it which ends the loop.
+    //We check to see if i is the same as a vowel by using the .includes method.
+    if (['a', 'e', 'i', 'o', 'u'].includes(string[i])){
```

Comment:
A for loop is a valid tool for this problem, but you could challenge yourself to do this using a HOF. Couple of things that you could look at doing here is declaring a variable for the vowels and also, if we return 0 when there is no vowel, we would be expecting 0 as well for if there is a vowel at index 0 / the first character.